src/fklearn/training/classification.py

```diff
@@ -313,12 +328,9 @@ def catboost_classification_learner(df: pd.DataFrame,
 
     def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
 
-        dtest = Pool(new_df[features].values, feature_names=list(map(str, features)),
-                     cat_features=cat_features)
-
-        pred = cbr.predict_proba(dtest)[:, 1]
+        pred = cbr.predict_proba(new_df[features])[:, 1]
```

Comment:
this should be inside the else in L325/R337. In the if case (MultiClass) the prediction is calculated twice, with overwriting.

Reply:
(just tried to have the full if/else in the comment but didnt manage to. The subsequent lines:
```
        pred = cbr.predict_proba(new_df[features])[:, 1]

        if params["objective"] == "MultiClass":
            pred = cbr.predict_proba(dtest)
            pred = cbr.predict_proba(new_df[features])
            col_dict = {prediction_column + "_" + str(key): value
                        for (key, value) in enumerate(pred.T)}
            col_dict.update({prediction_column: pred.argmax(axis=1)})
        else:
            col_dict = {prediction_column: pred}
```